### ISSUE TITLE
No subtitles

### ISSUE THREAD

**Issue by Gjur0** [cguZZman/plugin.onedrive#49]
Hi, I have still problem with plugin it doesn't pull the subtitle out of onedrive bussines. I did STRM file right, but it loads only integrated subtitle from mkv file. Srt file doesn't load even movie and subtitle have same name.

This is my log file: [https://paste.kodi.tv/pumicaliko.kodi](url)

![0](https://user-images.githubusercontent.com/13987932/106821757-0e7e4200-667e-11eb-8c12-744c76838f25.png)
![1](https://user-images.githubusercontent.com/13987932/106821758-0faf6f00-667e-11eb-872e-33782d69371a.png)
![2](https://user-images.githubusercontent.com/13987932/106821759-0faf6f00-667e-11eb-9c26-0a6ac43869e7.png)

**Comment by cguZZman** [comment]
Your log file does is not in debug mode. First enable the debug mode, then play your video again.
Do the playing by navigating thru the addon, not the library, so I can see more information in the debug log file.

**Comment by Gjur0** [comment]
Here is new log file:

[https://paste.kodi.tv/omutazisoc.kodi](url)

**Comment by cguZZman** [comment]
I can see the subtitle file in the list, but the API for business is not returning it when searching by name.
Do another test again by renaming one of your episodes. Just remove the spaces for now in both files, video and subtitle, and try playing it again.
Also post the debug log file again for this test.

**Comment by Gjur0** [comment]
New log file:

[https://paste.kodi.tv/ulupeziqok.kodi](url)

**Comment by cguZZman** [comment]
May be I was not clear in my previous request. The renaming I'd like to test is to remove all spaces in the file name, not just adding '-test' to the file name.
For example, can you rename your file `Star Trek Voyager - 1x04 - Time and Again-test.mkv` to `StarTrekVoyager-1x04-TimeandAgain-test.mkv`? (all spaces were removed). Or you can renamed as you want, but no spaces whatsoever.
Same for the subtitle.

**Comment by Gjur0** [comment]
Sorry, I read superficially, here is a new log file

https://paste.kodi.tv/egowowawur.kodi

**Comment by cguZZman** [comment]
Looks like spaces are not the problem. Last test: can you remove also the hyphen characters?

**Comment by Gjur0** [comment]
[https://paste.kodi.tv/otadojezet.kodi](url)

**Comment by cguZZman** [comment]
Thanks for you logs.
Looks like an issue with the search api for onedrive for business. The addon is doing the search of the subtitle correctly. It works correctly with a personal account.
I currently have no business account to check it out further. So there is not much I can do from here.

If anybody wants to temporary share an account it will be very welcome.

**Comment by marciomaiden** [comment]
> Thanks for your logs.
> Looks like an issue with the search api for onedrive for business. The addon is doing the search of the subtitle correctly. It works correctly with a personal account.
> I currently have no business account to check it out further. So there is not much I can do from here.
> 
> If anybody wants to temporary share an account it will be very welcome.

Do you need access to one account? I can give to you

**Comment by cguZZman** [comment]
Yeah, that would be great. Please send me the details to carlosguzmang at protonmail.com

**Comment by marciomaiden** [comment]
> Yeah, that would be great. Please send me the details to carlosguzmang at protonmail.com

If just send you the details

**Comment by cguZZman** [comment]
Received. Will recheck this issue with your account soon. Thanks

**Comment by felipefgf12** [comment]
I'm having the same problem with my business account.

**Comment by kampred** [comment]
is there any update for subtitle on OneDrive for Business account?

thanks

**Comment by felipefgf12** [comment]
> is there any update for subtitle on OneDrive for Business account?
> 
> thanks

**Comment by felipefgf12** [comment]
> It is confirmed that this is an issue in the API of OneDrive for Business, and specific to some business accounts.
> It is not an addon issue. I will post the link of the issue opened in OneDrive so you guys can track it there.

that's a bummer! could you share us the link now? thanks.

**Comment by oyuntuayC** [comment]
but I can still select the subtitle manually, is it possible to make this process automatically?
thanks

**Comment by jhayc1** [comment]
Reporting that this is still an issue (for OneDrive Business Accounts). Thanks.

**Comment by eras3r86** [comment]
> but I can still select the subtitle manually, is it possible to make this process automatically? thanks

how can you select it mannualy? If i browse to the movie folder no srt file is shown to download... thanks

**Comment by chonpc** [comment]
Today, 2024/aug/28, external subtitles *.srt cannot be played. Will there be any updates to correct it?

**Comment by alooo1414** [comment]
Hi everyone,  
I’ve faced similar challenges but found a workaround that works flawlessly for me on both PC and TV. I haven’t tested it on other platforms, but it should work there as well. Here's the solution:  

1. Sign in to your OneDrive: and go to the main menu.  
2. Select your preferred library (e.g., Movies, TV Shows) and click on it.  
3. Ensure you're at the root of the directory by navigating back using `..`.  
4. Go to `File` -> `+ Add Videos`.  
5. In the pop-up window, choose to add a video source and click `Browse`.  
6. Click `Add a network location` and fill in the following details:  
   - Protocol: **Web server directory (HTTP)**  
   - Server address: **localhost**  
   - Remote path: **source**  
   - Port: **8586**  
   - Leave the username and password empty.  
7. Click `OK` and add the new video source to the directory.  

Now, you should see the directory added to your library. When accessing videos:  
- Play the video from the **library** (not the add-on).  
- Select `Add Subtitle` and browse for the subtitle file.  
- You’ll find all subtitle files within the folder containing the video.  

This has been working smoothly for me without any issues so far. Hope this helps! 😊  

Let me know if you have any questions or feedback!
